src/app/api/cron/notifications/attendance/route.ts

```diff
@@ -0,0 +1,39 @@
+import { Tables } from "@/lib/db";
+import { getSupabaseClient } from "@/utils/supabase/supabaseClient";
+import { NextResponse } from "next/server";
+import { MemberRole, UserStatus } from "@/constant/constant";
+import * as cron from "node-cron";
+
+export async function GET() {
+  try {
+    cron.schedule("0 0 0 * * *", async () => {
+      const supabase = await getSupabaseClient();
+      const { data: users, error } = await supabase
+        .from(Tables.Members)
+        .select()
+        .or(
+          `role.eq.${MemberRole.Incharge},role.eq.${MemberRole.ShiftIncharge}`
+        )
+        .neq("status", UserStatus.Deactivated);
+      if (error) {
+        throw error;
+      }
+      for (const user of users) {
```

Comment:
We're doing series of db inseration. How about we around and use insertMany for one round trip to DB?

Reply:
fixed